### PR TITLE
Fix/Load all data for CSV

### PR DESCRIPTION
**Pull request by tsubik**
## Overview

CSVButton component has a new ability to take a function as data argument. That will help to load any data we want on demand so we can load the whole set even when we are using InfitieScroll or other paged list.

Other things:
- simple tests for CSVButton
- filter (csn, iba) is taken from the state now as the router updates the state after the route changes
- fixed some lint errors for About and Guidance pages

**[Story link](https://www.pivotaltracker.com/story/show/153173383)**

### Checklist

- [x] PR has a name that won't get you publicly shamed for vagueness
- [x] PR has a description that explains it
- [x] PR includes information for people to test it [e.g.: run `npm install`]
- [x] PR is not 2k commits long... please. :pray:

### CODE REVIEW

File: app/components/__tests__/AdvancedSearchPage.test.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const setup = () => {
   };
 
   const standard = {
-    filter: {},
+    filter: 'ibn',
```
Comment (simaob) [inline app/components/__tests__/AdvancedSearchPage.test.js]: This is a test file so it doesn't matter much, but the correct filter would be `iba`, just for information.

**Comment by simaob** [comment]
When on the `Sites` page I'm seeing a style issue when clicking on the selector, see attached screenshot:

screen shot 2017-12-05 at 17 59 59

I didn't see it on the deployed version, so I'm wondering if it was introduced in this PR.

**Comment by tsubik** [comment]
Weird.. I haven't touched any styles and locally works just fine.
image

**Comment by simaob** [comment]
Let me clear some cache and check again, see if it was just a glitch.

File: app/components/__tests__/AdvancedSearchPage.test.js (hunk 1 above)
Comment (tsubik) [reply app/components/__tests__/AdvancedSearchPage.test.js]: ok :+1:

**Comment by simaob** [comment]
Quite weird, I got the same behaviour locally on this branch in both Firefox and Chrome:

kapture 2017-12-05 at 18 18 17

**Pull request merged**